lib/beaker/hypervisor/openstack.rb

```diff
@@ -223,36 +233,9 @@ def provision
         end
 
         # Associate a public IP to the server
-        # Create if there are no floating ips available
-        #
-        # Do we already have an address?
-        @logger.debug vm.addresses
-        address=nil
-        begin
-          # Here we try and assign an address from a floating IP pool
-          # This seems to fail on some implementations (FloatingIpPoolNotFound)
-          ip.ip.gsub! '-','.'
-          ip.server = vm
-          address = ip.ip
-
-        rescue Fog::Compute::OpenStack::NotFound
-          # Here, we fail to just trying to use an address that's already assigned if there is one
-          # There may be better logic, but this worked in the original implementation
-          # There might be an argument for checking whether an address is reachable a la
-          # port_open? logic in host.rb but maybe race conditions
-
-          begin
-            if vm.addresses[@options[:openstack_network]]
-              address = vm.addresses[@options[:openstack_network]].map{ |network| network['addr'] }.first
-            end
-          rescue NoMethodError
-            @logger.debug "No current address retrievable from OpenStack data"
-          end
-
-        end
-
-        raise 'Could not find or assign an address to the instance' unless address
-        host[:ip] = address
+        ip.ip.gsub! '-','.'
```

Comment:
It seems that just after the `get_ip` call on line 199, we're setting the ip to have dashes rather than dots (or vice versa), and then setting them back here to be set correctly as a host attribute. Since we're assigning the changed IP address to the `hostname` variable, couldn't we just take off the `!` above, and then we'd be able to delete this line?